### DISCUSSION TITLE
Where do i use RouterLink and where NuxtLink?

### DISCUSSION THREAD

**Discussion by benjamin-ott** [shopware/frontends#62] [Q&A] [answered]
Hi,

I'm trying to find out if it is better to use NuxtLink for all links in the application instead of RouterLink.
The documentation of Nuxt states, that NuxtLink works both for internal and external links, so no problem there.

But which is better for performance? I found out NuxtLink prefetches resources for links which are in the visible viewport. So e.g. if i have a link to a wishlist page in an account popover in the header all wishlist resources would be prefetched if the user opens the popover. I think this could be a perfromance advantage right?

I also found another method in the vue-demo-store template where a button with a click event is used to push something to the router:
https://github.com/shopware/frontends/blob/main/templates/vue-demo-store/components/account/AccountMenu.vue#L77 

So is it a good idea to replace all RouterLink components with NuxtLink, or am i missing something?

**Comment by patzick** [comment, marked as answer]
Thanks for the question, @benjamin-ott 
As for Button-link - I created an issue for that #63 

As for the perspective of Nuxt Link - for the template part, it's better to use NuxtLink. The API is the same, but an additional prefetching advantage is good for perf. We use RouterLink in examples, etc., as then the code is more portable and does not require Nuxt.
I'll replace all links inside the Nuxt project, and we can recommend that to any Nuxt project as well :) especially that change between RouterLInk <-> NuxtLink is a simple code replacement with no additional work
